src/parser/CarsParser.java

```diff
@@ -0,0 +1,43 @@
+package parser;
+
+import static parser.AttributeValueParser.*;
+
+import entity.Car;
+import entity.Company;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class CarsParser {
+    private static List<Car> cars = new ArrayList<>();
+    private static final Logger LOGGER = Logger.getLogger(Company.class.getName());
+    private static final AttributeValueParser<Integer> idParser = new AttributeValueParser();
```

Comment:
What is the reason to have 2 parsers? just create only one and use preferred pattern to match.